agent/agent.go

```diff
@@ -78,6 +78,16 @@ func (a *Agent) Start(ctx context.Context) error {
 	return err
 }
 
+// Leave cleans up assignments from Swarm
```

Comment:
This needs a little more documentation about the meaning of Leave vs Close. This is pretty much permanent and may block on container shutdown.